docs/architecture/ErrorReporting.md

```diff
@@ -0,0 +1,144 @@
+
+                                    SendBox  ───────────────────────────► async   ─────────────────────────►  sendMessage
+                                        .sendMessageOnClick()              sendMessage()                       REST call
+                                                                                                                    │
+                                                                                                                    ▼
+                                    Failed promise error                  async                               sendMessage
+                                        ignored.         ◄───────────────  sendMessage()   ◄────────────────── REST call
+                                                                           returns failed promise              fails
+                                                                                    │
+                                                                            &&      │
+                                                                                    ▼
+                                                                           updates ChatState.errors
+                                                                                    │
+                                                                                    │
+                                                                                    ▼
+                                    adaptedErrorBarSelector ◄───────────── onStateChange() event
+                                        extracts error                         with new error
+                                        and updates error bar UI
+                                                │
+                                     &&         │
+                                                ▼
+                                    updates adapter.state.errors
+                                                │
+                                     &&         │
+                                                ▼
+    Custom error handling◄────────── triggers on('error', handler)
+    logic
+
+* `SendBox.sendMessageOnClick()` handler runs when the user clicks on a button, outside Reach component lifecycle.
+* The asynchronously returned error is dropped by the handler.
+* But the error is also persisted by the stateful client.
+  * The `ErrorBar` gets updated based on the state update.
+  * The `ChatComposite` also updates its own internal state and triggers an event that the application acts on.
+
+## UI errors
+
+These errors are _unexpected_ in that the UI library has a bug or is unable to handle an error from the browser. Neither the end-user, nor the surrounding application, is expected to act on the errors. As such, they aren't handled via the mechanism used for ACS backend errors.
```

Comment:
You mention below this that the special case should be handled by the application but here that "Neither the end-user, nor the surrounding application, is expected to act on the errors.". I think if cat2 is a catch-all there's likely going to be some errors an application will want to act on. Especially if we label local device permissions as a category 2

Reply:
Dropped the confusing classification of errors. Important point is errors that arise from the ACS clients are handled by the stateful client. Similarly, errors that arise behind the adapter API are expected to be handled by the adapters.

PTAL.